hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -152,98 +153,107 @@ private void addShutDownHook() {
 
-    HoodieLogBlock.HoodieLogBlockContentLocation logBlockContentLoc =
-        new HoodieLogBlock.HoodieLogBlockContentLocation(hadoopConf, logFile, contentPosition, contentLength, blockEndPos);
-
-    switch (Objects.requireNonNull(blockType)) {
-      case AVRO_DATA_BLOCK:
-        if (nextBlockVersion.getVersion() == HoodieLogFormatVersion.DEFAULT_VERSION) {
-          return HoodieAvroDataBlock.getBlock(content.get(), readerSchema, internalSchema);
-        } else {
-          return new HoodieAvroDataBlock(inputStream, content, readBlockLazily, logBlockContentLoc,
+          return new HoodieParquetDataBlock(inputStream, content, readBlockLazily, logBlockContentLoc,
               getTargetReaderSchemaForBlock(), header, footer, keyField);
-        }
-
-      case HFILE_DATA_BLOCK:
-        checkState(nextBlockVersion.getVersion() != HoodieLogFormatVersion.DEFAULT_VERSION,
-            String.format("HFile block could not be of version (%d)", HoodieLogFormatVersion.DEFAULT_VERSION));
-
-        return new HoodieHFileDataBlock(inputStream, content, readBlockLazily, logBlockContentLoc,
-            Option.ofNullable(readerSchema), header, footer, enableRecordLookups, logFile.getPath());
 
-      case PARQUET_DATA_BLOCK:
-        checkState(nextBlockVersion.getVersion() != HoodieLogFormatVersion.DEFAULT_VERSION,
-            String.format("Parquet block could not be of version (%d)", HoodieLogFormatVersion.DEFAULT_VERSION));
+        case DELETE_BLOCK:
+          return new HoodieDeleteBlock(content, inputStream, readBlockLazily, Option.of(logBlockContentLoc), header, footer);
 
-        return new HoodieParquetDataBlock(inputStream, content, readBlockLazily, logBlockContentLoc,
-            getTargetReaderSchemaForBlock(), header, footer, keyField);
+        case COMMAND_BLOCK:
+          return new HoodieCommandBlock(content, inputStream, readBlockLazily, Option.of(logBlockContentLoc), header, footer);
 
-      case DELETE_BLOCK:
-        return new HoodieDeleteBlock(content, inputStream, readBlockLazily, Option.of(logBlockContentLoc), header, footer);
+        case CDC_DATA_BLOCK:
+          return new HoodieCDCDataBlock(inputStream, content, readBlockLazily, logBlockContentLoc, readerSchema, header, keyField);
 
-      case COMMAND_BLOCK:
-        return new HoodieCommandBlock(content, inputStream, readBlockLazily, Option.of(logBlockContentLoc), header, footer);
-
-      case CDC_DATA_BLOCK:
-        return new HoodieCDCDataBlock(inputStream, content, readBlockLazily, logBlockContentLoc, readerSchema, header, keyField);
-
-      default:
-        throw new HoodieNotSupportedException("Unsupported Block " + blockType);
+        default:
+          throw new HoodieNotSupportedException("Unsupported Block " + blockType);
+      }
+    } catch (IOException | CorruptedLogFileException | IllegalArgumentException e) {
+      // check for corrupt block
+      inputStream.seek(blockStartPos);
+      if (isBlockCorrupted(blockSize)) {
```

Comment:
may I know why do we check for block corrupt here again? can't we assume if something is thrown, its is indeed corrupt?

Reply:
If we are here due to IOException then it may or may not be a corrupt block. IOException can occur for various reasons.

Reply:
Can we remove the catch for `IllegalArgumentException e` and can we check that the exception is a `IOException` before we do another check for `isBlockCorrupted(blockSize)` ?

Reply:
Moved the various exceptions to their own catch blocks to make the handling clear.

Reply:
Moved the various exceptions to their own catch blocks to make the handling clear.